script.js

```diff
@@ -1,4 +1,230 @@
+};
+
+//sort largest to smallest number
+var sortBigToSmall = function (input) {
+
+  for (i = 0; i < input.length; i += 1) {
+    for (j = 0; j < (input.length - i - 1); j ++) {
+      if (input[j] < input[j+1]) {
+        var temp = input[j];
+        input[j] = input[j+1];
+        input[j+1] = temp;
+      } 
+    }
+      console.log(input)
+      console.log(j)
+      console.log(i)
+  }
+  return null;
+}
+
+//Converts array of numbers to a number
+var convertArrayToNumber = function (input) {
+  var runDown = chosenDiceRolls - 1;
+  var counter = 0;
+
+  while (runDown >= 0) {
+    var number = input[counter] * Math.pow(10, runDown);
+    console.log(number);
+    playerInPlay.tally += number;
+    console.log(playerInPlay.tally);
+    runDown -= 1;
+    counter += 1;
+  }
+  return playerInPlay.tally;
+};
+
+//Compares the score 
+var tallyComparison = function () {
+  var i = 0;
+  roundWinner = playersList[i];
+
+  while (i < playersList.length - 1) {
+    if (roundWinner.tally > playersList[i+1].tally) {
+      roundWinner = roundWinner;
+    } else {
+      roundWinner = playersList[i+1];
+    }
+    i +=1;
+  }
+
+  roundWinner.score +=1;
+  console.log (playersList.score);
+  return roundWinner;
+};
+
+//Find out who is leading 
+var scoreComparison = function () {
+  var i = 0;
+  overallWinner = playersList[i];
+
+  while (i < playersList.length - 1) {
+    if (overallWinner.score > playersList[i+1].score) {
+      overallWinner = overallWinner;
+    } else {
+      overallWinner = playersList[i+1];
+    }
+    i +=1;
+  }
+
+  return overallWinner;
+};
+
+//reset
+var resetGame = function () {
+  var counter = 0;
+
+  while (counter < playersList.length) {
+    playersList[counter].hand = [];
+    playersList[counter].tally = 0;
+
+    counter += 1;
+  }
+  gameState = DICE_ROLL;
+  return null;
+};
+
+//main
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  //check game state and player before start
+  console.log(gameState);
+
+  if (gameState == START) {
+    gameState = SET_PLAYERS;
+    console.log(gameState);
+    return `Begin by choosing the number of players playing the game.`;
+  }
+
+  if (gameState == SET_PLAYERS) {
+    createPlayersList(input);
+    gameState = SET_ROLLS;
+    console.log(playersList);
+    return `You have chosen to play with ${input} players. Now choose the number of dice to play with.`;
+  }
+  
+  if (gameState == SET_ROLLS) {
+    chosenDiceRolls = input;
+    gameState = DICE_ROLL;
+    return `You have chosen to play with ${chosenDiceRolls} dice. Time to rock and roll (pun intended!). Click Submit for all players to roll the dice.`;
+  }
+
+  if (gameState == DICE_ROLL) {
+    var playerCounter = 0;
+    var outputStatement = 'Let the games begin (again).<br>';
+
+    while (playerCounter < playersList.length) {
+      playerInPlay = playersList[playerCounter];
+
+      rollDiceForPlayer();
+      var copiedHand = [...playerInPlay.hand];
```

Comment:
Nice to see that you're copying an array with this method! There are other ways to do it as well, like with `Array.slice`, but this is the method I prefer the most!